admin/automation/process.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import logging
+import os
+import sys
+from enum import Enum
+from typing import List, Optional
+
+from pydantic import BaseModel, Field, ValidationError
+
+
+class Cli:
+    def __init__(self) -> None:
+        self.parser = argparse.ArgumentParser(
+            description="""
+            Extract useful information from processed grant files
+            """
+        )
+        self.parser.add_argument(
+            "input",
+            type=dir_path,
+            help="Input directory, containing grant files",
+        )
+        self.parser.add_argument(
+            "-d",
+            "--debug",
+            action="store_true",
+            help="Print debugging information",
+        )
+        self.args = self.parser.parse_args()
+
+
+def dir_path(string):
+    if os.path.isdir(string):
+        return string
+    else:
+        raise NotADirectoryError(string)
+
+
+class Status(str, Enum):
+    SELECTED = "Selected"
+    RUNNING = "Running"
+    COMPLETE = "Complete"
+
+
+class Contact(BaseModel):
+    name: str
+    email: str
+    organisationName: str
+
+
+class Websites(BaseModel):
+    website: List[str]
+
+
+class Webpage(BaseModel):
+    sitename: Optional[str] = Field(default=None)
+    summary: str
+
+
+class PropertiesField(BaseModel):
+    webpage: Webpage
+
+
+class ProposalField(BaseModel):
+    websites: Websites
+    contact: Contact
+
+
+class Proposal(BaseModel):
```

Comment:
```suggestion
class Subgrant(BaseModel):
```

Same as https://github.com/ngi-nix/summer-of-nix/pull/30/files#r1934852803